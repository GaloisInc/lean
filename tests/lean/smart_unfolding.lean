constant n : nat
#reduce n + (nat.succ n)

set_option type_context.smart_unfolding false
#reduce n + (nat.succ n)
