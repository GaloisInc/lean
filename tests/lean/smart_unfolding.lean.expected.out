nat.succ (nat.add n n)
nat.succ
  ((nat.rec ⟨λ (a : ℕ), a, punit.star⟩
      (λ (n : ℕ)
       (ih : pprod (ℕ → ℕ) (nat.rec punit (λ (n : ℕ) (ih : Type), pprod (pprod (ℕ → ℕ) ih) punit) n)),
         ⟨λ (a : ℕ), nat.succ (ih.fst a), ⟨ih, punit.star⟩⟩)
      n).fst
     n)
